src/js/tabs/motors.js

```diff
@@ -681,56 +896,93 @@ TABS.motors.initialize = function (callback) {
             }
 
-            // servo indicators are still using old (not flexible block scale), it will be changed in the future accordingly
-            for (let i = 0; i < FC.SERVO_DATA.length; i++) {
-                const data = FC.SERVO_DATA[i] - 1000,
-                marginTop = blockHeight - (data * (blockHeight / 1000)).clamp(0, blockHeight),
-                height = (data * (blockHeight / 1000)).clamp(0, blockHeight),
-                color = parseInt(data * 0.009);
-
-                $(`.servo-${i} .label`, servos_wrapper).text(FC.SERVO_DATA[i]);
-                $(`.servo-${i} .indicator`, servos_wrapper).css({
-                    'margin-top' : `${marginTop}px`,
-                    'height' : `${height}px`,
-                    'background-color' : `rgba(255,187,0,1${color})`,
-                });
-            }
             //keep the following here so at least we get a visual cue of our motor setup
             update_arm_status();
 
             if (previousArmState != self.armed) {
                 console.log('arm state change detected');
 
-                $('#motorsEnableTestMode').change();
+                motorsEnableTestModeElement.change();
             }
         }
 
+        $('a.save').on('click', function() {
+            mspHelper.setArmingEnabled(false, true);
+
+            // gather data that doesn't have automatic change event bound
+            FC.MOTOR_CONFIG.minthrottle = parseInt($('input[name="minthrottle"]').val());
+            FC.MOTOR_CONFIG.maxthrottle = parseInt($('input[name="maxthrottle"]').val());
+            FC.MOTOR_CONFIG.mincommand = parseInt($('input[name="mincommand"]').val());
+
+            if (semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_42)) {
+                FC.MOTOR_CONFIG.motor_poles = parseInt($('input[name="motorPoles"]').val());
+            }
+
+            FC.MOTOR_3D_CONFIG.deadband3d_low = parseInt($('input[name="3ddeadbandlow"]').val());
+            FC.MOTOR_3D_CONFIG.deadband3d_high = parseInt($('input[name="3ddeadbandhigh"]').val());
+            FC.MOTOR_3D_CONFIG.neutral = parseInt($('input[name="3dneutral"]').val());
+
+            FC.PID_ADVANCED_CONFIG.fast_pwm_protocol = parseInt(escProtocolElement.val() - 1);
+            FC.PID_ADVANCED_CONFIG.use_unsyncedPwm = unsyncedPWMSwitchElement.is(':checked') ? 1 : 0;
+            FC.PID_ADVANCED_CONFIG.motor_pwm_rate = parseInt($('input[name="unsyncedpwmfreq"]').val());
+            FC.PID_ADVANCED_CONFIG.digitalIdlePercent = parseFloat($('input[name="digitalIdlePercent"]').val());
+
+            if (semver.gte(FC.CONFIG.apiVersion, "1.25.0") && semver.lt(FC.CONFIG.apiVersion, API_VERSION_1_41)) {
+                FC.PID_ADVANCED_CONFIG.gyroUse32kHz = $('input[id="gyroUse32kHz"]').is(':checked') ? 1 : 0;
+            }
+
+            const saveSettings = async () => {
+                try {
+                    await MSP.promise(MSPCodes.MSP_SET_FEATURE_CONFIG, mspHelper.crunch(MSPCodes.MSP_SET_FEATURE_CONFIG));
```

Comment:
Using a block of consecutive awaits on individual promises is completely against the nature of promises, and will likely lead to problems and a frozen UI due to the main thread being blocked for a long time.
Instead, use promise chaining in the way that MSP promises are used in other places in the application, like `src/js/tabs/pid_tuning.js` to get the MSP communication done in a background thread.

Reply:
$('a.save').on('click', function() {
            mspHelper.setArmingEnabled(false, true);

This is stopping the motors.
Have tested save and reboot many times but will adapt the method used in pid_tuning...

Reply:
Done.